src/components/date-picker/date-picker-date-utils.ts

```
@@ -100,78 +119,103 @@ export function generateDatePickerColumns(
-  }
+  const finalColumns = columns || defaultColumns
+
+  const validColumns = finalColumns.filter(columnType => {
+    const columnPrecision = columnToPrecisionMap[columnType]
+    return rank >= precisionRankRecord[columnPrecision]
+  })
+
+  validColumns.forEach(columnType => {
+    switch (columnType) {
+      case YEAR_COLUMN:
+        const lower = minYear
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)